snap/snapcraft.yaml

```diff
@@ -0,0 +1,278 @@
+      - -DPYTHON_EXECUTABLE=/usr/bin/python3
+      - -DPYTHON_INCLUDE_DIR=/usr/include/python3.8
+      - -DPYTHON_LIBRARY=/usr/lib/$SNAPCRAFT_ARCH_TRIPLET/libpython3.8.so
+      - -DFREECAD_USE_EXTERNAL_ZIPIOS=ON
+      - -DFREECAD_USE_PYBIND11=ON
+      - -DFREECAD_USE_QT_FILEDIALOG=OFF
+      - -DBUILD_FLAT_MESH=ON
+    build-packages:
+      - g++
+      - git
+      - libboost-all-dev
+      - libsimage-dev # optional
+      - libspnav-dev # optional
+      - libeigen3-dev
+      - libgts-bin
+      - libgts-dev
+      - libkdtree++-dev
+      - libmedc-dev
+      - libopencv-dev
+      - libproj-dev
+      - libx11-dev
+      - libxerces-c-dev
+      - libzipios++-dev
+      - swig
+      - python3-dev
+      - libocct-data-exchange-dev
+      - libocct-draw-dev
+      - libocct-foundation-dev
+      - libocct-modeling-algorithms-dev
+      - libocct-modeling-data-dev
+      - libocct-ocaf-dev
+      - libocct-visualization-dev
+      - occt-draw
+      - libvtk7-dev
+      - libpyside2-dev
+      - libshiboken2-dev
+      - pybind11-dev
+    stage-packages:
+      - libaec0
+      - libboost-filesystem1.71.0
+      - libboost-program-options1.71.0
+      - libboost-python1.71.0
+      - libboost-regex1.71.0
+      - libboost-system1.71.0
+      - libboost-thread1.71.0
+      - libboost-date-time1.71.0 
+      - libhdf5-openmpi-103
+      - libhwloc15
+      - libilmbase24
+      - libjxr0
+      - libmedc11
+      - libmed11
+      - libopenexr24
+      - libopenmpi3
+      - libpython3.8
+      - libpython3.8-minimal
+      - libpython3.8-stdlib
+      - libraw19
+      - libspnav0
+      - libsz2
+      - libxerces-c3.2
+      - libzipios++0v5
+      - python3-tk # FEM
+      - python3-yaml # FEM
+      - python3-numpy
+      - python3-matplotlib
+      - python3-six
+      - python3-pyparsing
+      - python3-setuptools-scm
+      - python3-cycler
+      - python3-dateutil
+      - python3-git
+      - python3-ply # OpenSCAD
+      - python3-markdown # Addon manager
+      - python3-pyside2.qtcore
+      - python3-pyside2.qtgui
+      - python3-pyside2.qtsvg
+      - python3-pyside2.qtwidgets
+      - python3-pyside2.qtnetwork
+      - graphviz
+      - calculix-ccx # FEM
+      - libfreeimage3
+      - libocct-data-exchange-7.5
+      - libocct-foundation-7.5
+      - libocct-modeling-algorithms-7.5
+      - libocct-modeling-data-7.5
+      - libocct-ocaf-7.5
+      - libocct-visualization-7.5
+      - libtbb2
+      - libvtk7.1p
+      - elmerfem-csc # FEM
+      - openscad  # OpenSCAD
+    override-build: |
+      kde_sdk_dir="/snap/kde-frameworks-5-qt-5-15-3-core20-sdk/current"
+      mkdir -p /etc/xdg/qtchooser
+      cp "$kde_sdk_dir/etc/xdg/qtchooser/default.conf" "/etc/xdg/qtchooser/default.conf"
+      snapcraftctl build
+      cd $SNAPCRAFT_PART_SRC
+      iso_date=$(date -I)
+      git_hash=$(git rev-parse --short=8 HEAD)
+      version="$iso_date-g$git_hash"
+      snapcraftctl set-version "$version"
```

Comment:
The versioning scheme is: "$iso_date-g$git_hash", e.g.  `2022-01-09-g0e45378e`. You use a similar, date-based scheme for your appimages. Do you want to stick to that?

Reply:
I think the scheme you use is better.